docs/api/v3.rst

```diff
@@ -0,0 +1,558 @@
+                },
+                {
+                    "href": "/api/v3/project/pip/",
+                    "rel": "project",
+                    "type": "GET"
+                },
+                {
+                    "href": "/api/v3/project/pip/version/latest/",
+                    "rel": "version",
+                    "type": "GET"
+                }
+        }
+
+
+    :>json integer id: The ID of the build
+    :>json string date: The ISO-8601 datetime of the build.
+    :>json integer length: The length of the build in seconds.
+    :>json string type: The type of the build (one of "html", "pdf", "epub")
+    :>json string state: The state of the build (one of "triggered", "building", "installing", "cloning", or "finished")
+    :>json string state_display: The state of the build to be shown to the user
+    :>json boolean success: Whether the build was successful
+    :>json string error: An error message if the build was unsuccessful
+    :>json string commit: A version control identifier for this build (eg. the commit hash)
+    :>json string docs_url: The canonical URL of the build docs
+    :>json string builder: The hostname server that built the docs
+    :>json array links: Array with HEATEOAS_ links to retrieve related information
+
+    :statuscode 200: Success
+    :statuscode 404: There is no ``Build`` with this ID
+
+
+Build triggering
+++++++++++++++++
+
+.. TODO:
+
+   This endpoint may be under Project section
+
+.. http:post::  /api/v3/project/(string:slug)/builds/
+
+    Trigger a new build for this project.
+
+    **Example request**:
+
+    .. sourcecode:: js
+
+        {
+            "version": "latest",
+        }
+
+    **Example response**:
+
+    *See Build details*
+
+    :statuscode 201: Created sucessfully
+    :statuscode 400: Some field is invalid
+    :statuscode 401: Not valid permissions
+
+
+Footer
+~~~~~~
+
+
+.. http:get::  /api/v3/footer/(string:project_slug)/
+
+    Retrieve footer data for the project.
+
+    **Example response**:
+
+    .. sourcecode:: js
+
+        {
+            "version": {VERSION DETAIL},
+            "project": {PROJECT DETAIL},
+            "sphinx": {
+                "html_theme": "sphinx_rtd_theme",
+                "source_suffix": ".rst"
+            },
+            "analytics": {
+                "user_analytics_code": "UA-00000",
+                "global_analytics_code": "UA-00001"
+            },
+            "vcs": {
+                "type": "github"
+                "user": "pypa",
+                "repo": "pip",
+                "commit": "3a6b3995c141c0888af6591a59240ba5db7d9914",
+                "version": "18.1",
+                "display": true,
+                "conf_py_path": "/docs/conf.py"
+            },
+            "meta": {
+                "API_HOST": "readthedocs.org",
+                "MEDIA_URL": "https://media.readthedocs.org",
+                "PRODUCTION_DOMAIN": "readthedocs.io",
+                "READTHEDOCS": true
```

Comment:
Those are the only fields with uppercase keys

Reply:
Yes, these come from the template context and I'm keeping compatibility/nomenclature from there.